packages/protocol/src/action-protocol/element-hover.ts

```diff
@@ -13,47 +13,79 @@
 import { SModelRootSchema } from './model-structure';
-import { Bounds } from './types';
 
 /**
  * Triggered when the user hovers the mouse pointer over an element to get a popup with details on that element.
  * This action is sent from the client to the server. The response is a `SetPopupModelAction`.
+ * The corresponding namespace declares the action kind as constant and offers helper functions for type guard checks
+ * and creating new `RequestPopupModelActions`.
  */
-export class RequestPopupModelAction implements RequestAction<SetPopupModelAction> {
-    static readonly KIND = 'requestPopupModel';
-    readonly kind = RequestPopupModelAction.KIND;
+export interface RequestPopupModelAction extends RequestAction<SetPopupModelAction>, sprotty.RequestPopupModelAction {
+    /**
+     * The unique action kind.
+     */
+    kind: typeof RequestPopupModelAction.KIND;
 
-    constructor(public readonly elementId: string, public readonly bounds: Bounds, public readonly requestId = '') {}
+    /**
+     * The identifier of the elements for which a popup is requested.
+     */
+    elementId: string;
 
-    /** Factory function to dispatch a request with the `IActionDispatcher` */
-    static create(elementId: string, bounds: Bounds): RequestAction<SetPopupModelAction> {
-        return new RequestPopupModelAction(elementId, bounds, generateRequestId());
-    }
+    /**
+     * The bounds.
+     */
+    bounds: Bounds;
 }
 
-export function isRequestPopupModelAction(action: any): action is RequestPopupModelAction {
-    return (
-        isActionKind(action, RequestPopupModelAction.KIND) &&
-        isString(action, 'elementId') &&
-        isObject(action, 'bounds') &&
-        isString(action, 'requestId')
-    );
+export namespace RequestPopupModelAction {
+    export const KIND = 'requestPopupModel';
+
+    export function is(object: any): object is RequestPopupModelAction {
+        return RequestAction.hasKind(object, KIND) && hasStringProp(object, 'elementId') && hasObjectProp(object, 'bounds');
+    }
+
+    export function create(options: { elementId: string; bounds: Bounds; requestId?: string }): RequestPopupModelAction {
+        return {
+            kind: KIND,
+            requestId: '',
+            ...options
+        };
+    }
 }
 
 /**
  * Sent from the server to the client to display a popup in response to a RequestPopupModelAction. This action can also be used to remove
  * any existing popup by choosing EMPTY_ROOT as root element.
+ * The corresponding namespace declares the action kind as constant and offers helper functions for type guard checks
+ * and creating new `SetPopupModelActions`.
  */
-export class SetPopupModelAction implements ResponseAction {
-    static readonly KIND = 'setPopupModel';
-    readonly kind = SetPopupModelAction.KIND;
+export interface SetPopupModelAction extends ResponseAction, sprotty.SetPopupModelAction {
+    /**
+     * The unique action kind.
+     */
```

Comment:
I'd spare such comments, also if it hurts to be "incomplete". But imho they just add noise and add no information.